e2e/src/test/java/io/aiven/kafka/tieredstorage/e2e/SingleBrokerTest.java

```diff
@@ -0,0 +1,631 @@
+            assertThat(record.offset()).isEqualTo(newStartOffset);
+        }
+    }
+
+    @Test
+    @Order(4)
+    void remoteCleanupDueToRetention() throws Exception {
+        maybeFailFast();
+
+        // Collect all remote segments, as after changing retention, all should be deleted.
+        final var remoteSegmentsBefore = remoteLogMetadataTracker.remoteSegments();
+        final var segmentsToBeDeleted = Stream.concat(
+            remoteSegmentsBefore.get(t0p0).stream(),
+            remoteSegmentsBefore.get(t0p1).stream()
+        ).collect(Collectors.toList());
+
+        LOG.info("Forcing cleanup by setting bytes retention to 1");
+
+        final var alterConfigs = List.of(new AlterConfigOp(
+            new ConfigEntry("retention.bytes", "1"), AlterConfigOp.OpType.SET));
+        adminClient.incrementalAlterConfigs(Map.of(
+            new ConfigResource(ConfigResource.Type.TOPIC, TOPIC_0), alterConfigs
+        )).all().get(5, TimeUnit.SECONDS);
+
+        LOG.info("Starting cleanup validation");
+
+        try (final var consumer = new KafkaConsumer<>(Map.of(
+            "bootstrap.servers", kafka.getBootstrapServers(),
+            "auto.offset.reset", "earliest"
+        ), new ByteArrayDeserializer(), new ByteArrayDeserializer())) {
+
+            // Get earliest offset available locally
+            final long newStartOffset = localLogFiles(TP_0_0).stream()
+                .mapToLong(f -> Long.parseLong(f.getName().replace(".log", "")))
+                .max()
+                .getAsLong();
+            // and wait til expected earliest offset is in place
+            await()
+                .pollInterval(Duration.ofSeconds(5))
+                .atMost(Duration.ofSeconds(30))
+                .until(() -> {
+                    final var beginningOffset = consumer.beginningOffsets(List.of(TP_0_0)).get(TP_0_0);
+                    LOG.info("Beginning offset found {}, expecting {}", beginningOffset, newStartOffset);
+                    return beginningOffset.equals(newStartOffset);
+                });
+
+            final Map<TopicPartition, Long> endOffsets = consumer.endOffsets(List.of(TP_0_0));
+            assertThat(endOffsets.get(TP_0_0)).isEqualTo(RECORDS_TO_PRODUCE + 1);
+
+            // TODO check for EARLIEST_LOCAL_TIMESTAMP when available in client
+
+            consumer.assign(List.of(TP_0_0));
+            consumer.seek(TP_0_0, 0);
+
+            final ConsumerRecord<byte[], byte[]> record = consumer.poll(Duration.ofSeconds(1)).records(TP_0_0).get(0);
+            assertThat(record.offset()).isEqualTo(newStartOffset);
+
+            remoteLogMetadataTracker.waitUntilSegmentsAreDeleted(segmentsToBeDeleted);
+            await()
+                .between(Duration.ofSeconds(1), Duration.ofSeconds(30))
+                .pollInterval(Duration.ofSeconds(1))
+                .until(() -> assertNoTopicDataOnTierStorage(t0p0.topic(), t0p0.topicId()));
```

Comment:
This was the last bit that seemed to be flaky. Before it was only checked once and there was no guarantee that files were removed. Now, it's checked for a bit. Builds seem to be more stable now.